lib/smarter_csv/auto_detection.rb

```diff
@@ -19,7 +19,10 @@ def guess_column_separator(filehandle, options)
       count.times do
         line = readline_with_counts(filehandle, options)
         delimiters.each do |d|
-          candidates[d] += line.scan(d).count
+          # Count only non-quoted occurrences of the delimiter
+          non_quoted_text = line.split(/"[^"]*"|'[^']*'/).join
```

Comment:
`:quote_char` can be passed in by the user. 
It would be better if this would use `options[:quote_char]`

Reply:
@tilo Oops, missed that. Implemented the change, thank you!